### PR TITLE
SPLICE-1141 Allow for empty directories for ORC and Parquet external …

### PR DESCRIPTION
**Pull request by jleach4**
…tables.

Returning empty set when there are no files in the directory for PARQUET and ORC.

### CODE REVIEW

File: hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java [hunk 1]
```diff
@@ -33,6 +35,14 @@
...
     public static void cleanoutDirectory() {
+        try {
+            File file = new File(getExternalResourceDirectory());
+            if (file.exists())
+                FileUtils.deleteDirectory(new File(getExternalResourceDirectory()));
+            file.mkdir();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
     }
```
Comment (jfilali) [inline hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java]: why are checking for the existence of the file after you catch the exception ? why not checking before you try to read or create a empty parquet file ? Is this for performance reason ?

File: hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java [hunk 2]
```diff
@@ -306,6 +313,27 @@ public Partitioner getPartitioner(DataSet<LocatedRow> dataSet, ExecRow template,
...
+        // Cannot Infer Schema, Argh
+        if (e instanceof AnalysisException && e.getMessage() != null && e.getMessage().startsWith("Unable to infer schema")) {
+            // Lets check if there are existing files...
+            FileInfo fileInfo = ImportUtils.getImportFileInfo(location);
+            if (fileInfo.fileCount() == 0) // Handle Empty Directory
+                return getEmpty();
+        }
+        throw e;
+    }
+
```
Comment (jfilali) [inline hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java]: why are you checking for the existence of the file after you catch the exception ? why not checking before you try to read or create a empty parquet file ? Is this for performance reason ?

Comment (jleach4) [inline hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java]: I figured an empty table would be a special case.  I do not want to recurse the files unless I have to (exception).

**Comment by genedavis** [comment]
+1

**Pull request merged**